### PR TITLE
refactor: move funboxes to a shared package (@miodec)

### PR DESCRIPTION
**Pull request by Miodec**

### CODE REVIEW

File: frontend/src/ts/test/funbox/funbox-functions.ts [hunk 1]
```diff
@@ -0,0 +1,633 @@
...
+import * as LayoutfluidFunboxTimer from "./layoutfluid-funbox-timer";
+import * as KeymapEvent from "../../observables/keymap-event";
+import * as MemoryTimer from "./memory-funbox-timer";
+import { getPoem } from "../poetry";
+import * as JSONData from "../../utils/json-data";
+import { getSection } from "../wikipedia";
+import * as WeakSpot from "../weak-spot";
+import * as IPAddresses from "../../utils/ip-addresses";
+
+//todo figure out how to connect these frontend function names with the ones defined in the shared package
```
Comment (fehmer) [inline frontend/src/ts/test/funbox/funbox-functions.ts:25]: obsolete?

File: frontend/src/ts/ready.ts [hunk 2]
```diff
@@ -21,9 +20,9 @@ $((): void => {
   $("body").css("transition", "background .25s, transform .05s");
   MerchBanner.showIfNotClosedBefore();
   setTimeout(() => {
-    FunboxList.get(Config.funbox).forEach((it) =>
-      it.functions?.applyGlobalCSS?.()
-    );
+    for (const fb of Funbox.getActive()) {
+      fb.functions?.applyConfig?.();
```
Comment (fehmer) [inline frontend/src/ts/ready.ts:24]: `applyGlobalCSS` ?

File: frontend/src/ts/test/funbox/funbox.ts [hunk 3]
```diff
@@ -1,589 +1,109 @@
...
-      const nextChar = charDistribution.randomChar();
-      if (nextChar === " ") {
-        // A space marks the end of the word, so stop generating and return.
-        break;
-      }
-      word += nextChar;
-    }
-    return word;
-  }
+export function getActive(): FunboxMetadataWithFunctions[] {
```
Comment (fehmer) [inline frontend/src/ts/test/funbox/funbox.ts:61]: nitpick, if a function returns an array/list I'd use plural naming, e.g.  `getActives`

File: frontend/src/ts/test/funbox/funbox-functions.ts [hunk 4]
```diff
@@ -0,0 +1,633 @@
...
+        // A space marks the end of the word, so stop generating and return.
+        break;
+      }
+      word += nextChar;
+    }
+    return word;
+  }
+}
+
+const list: Partial<Record<FunboxName, FunboxFunctions>> = {
```
[the remark this answers is not in the document]
Comment (Miodec) [reply frontend/src/ts/test/funbox/funbox-functions.ts:155]: discussed on discord

**Review by fehmer — APPROVED** [review]
lgtm

**Pull request merged**